From 83c4d9afb21f6ac8956e0643cbbd6c823b700ac9 Mon Sep 17 00:00:00 2001
From: Dinolek <git@dinolek.me>
Date: Sat, 12 Jul 2025 21:57:07 +0200
Subject: [PATCH 104/115] arm64: dts: mediatek: mt6785-xiaomi-begonia: Enable
 touchscreen

Define shared SPI touchscreen wiring and enable the CSOT and Tianma nt36672a touchscreen firmware variants.

Signed-off-by: Dinolek <git@dinolek.me>
---
 .../mt6785-xiaomi-begonia-common.dtsi         | 48 +++++++++++++++++++
 .../mediatek/mt6785-xiaomi-begonia-csot.dts   |  5 ++
 .../mediatek/mt6785-xiaomi-begonia-tianma.dts |  5 ++
 3 files changed, 58 insertions(+)

diff --git a/arch/arm64/boot/dts/mediatek/mt6785-xiaomi-begonia-common.dtsi b/arch/arm64/boot/dts/mediatek/mt6785-xiaomi-begonia-common.dtsi
index 2d30f6df3..2887db82a 100644
--- a/arch/arm64/boot/dts/mediatek/mt6785-xiaomi-begonia-common.dtsi
+++ b/arch/arm64/boot/dts/mediatek/mt6785-xiaomi-begonia-common.dtsi
@@ -242,6 +242,28 @@ pins-reset {
 		};
 	};
 
+	spi5_pins: spi5-pins {
+		pins-bus {
+			pinmux = <PINMUX_GPIO37__FUNC_SPI5_A_MI>,
+				 <PINMUX_GPIO36__FUNC_SPI5_A_CSB>,
+				 <PINMUX_GPIO35__FUNC_SPI5_A_MO>,
+				 <PINMUX_GPIO38__FUNC_SPI5_A_CLK>;
+			bias-disable;
+		};
+	};
+
+	touchscreen_pins: touchscreen-pins {
+		pins-interrupt {
+			pinmux = <PINMUX_GPIO1__FUNC_GPIO1>;
+			bias-pull-up;
+			input-enable;
+		};
+
+		pins-reset {
+			pinmux = <PINMUX_GPIO2__FUNC_GPIO2>;
+		};
+	};
+
 	volume_button_pins: volume-button-pins {
 		pins-volume {
 			pinmux = <PINMUX_GPIO59__FUNC_KPCOL0>,
@@ -251,6 +273,32 @@ pins-volume {
 	};
 };
 
+&spi5 {
+	mediatek,pad-select = <0>;
+
+	pinctrl-names = "default";
+	pinctrl-0 = <&spi5_pins>;
+
+	status = "okay";
+
+	touchscreen: touchscreen@0 {
+		compatible = "novatek,nt36672a-ts";
+		reg = <0>;
+
+		interrupts-extended = <&pio 1 IRQ_TYPE_EDGE_RISING>;
+		reset-gpios = <&pio 2 GPIO_ACTIVE_HIGH>;
+		panel = <&panel>;
+
+		spi-cs-setup-delay-ns = <20>;
+		spi-max-frequency = <5000000>;
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&touchscreen_pins>;
+
+		status = "disabled";
+	};
+};
+
 &pmic {
 	interrupts-extended = <&pio 193 IRQ_TYPE_LEVEL_HIGH>;
 
diff --git a/arch/arm64/boot/dts/mediatek/mt6785-xiaomi-begonia-csot.dts b/arch/arm64/boot/dts/mediatek/mt6785-xiaomi-begonia-csot.dts
index eb6423f62..f7f9e8852 100644
--- a/arch/arm64/boot/dts/mediatek/mt6785-xiaomi-begonia-csot.dts
+++ b/arch/arm64/boot/dts/mediatek/mt6785-xiaomi-begonia-csot.dts
@@ -16,3 +16,8 @@ &panel {
 	compatible = "xiaomi,begonia-csot-nt36672a", "novatek,nt36672a";
 	status = "okay";
 };
+
+&touchscreen {
+	firmware-name = "novatek/nt36672a_begonia_csot.bin";
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/mediatek/mt6785-xiaomi-begonia-tianma.dts b/arch/arm64/boot/dts/mediatek/mt6785-xiaomi-begonia-tianma.dts
index 337959dcb..8084ccfe9 100644
--- a/arch/arm64/boot/dts/mediatek/mt6785-xiaomi-begonia-tianma.dts
+++ b/arch/arm64/boot/dts/mediatek/mt6785-xiaomi-begonia-tianma.dts
@@ -16,3 +16,8 @@ &panel {
 	compatible = "xiaomi,begonia-tianma-nt36672a", "novatek,nt36672a";
 	status = "okay";
 };
+
+&touchscreen {
+	firmware-name = "novatek/nt36672a_begonia_tianma.bin";
+	status = "okay";
+};
-- 
2.56.0

